src/libsyntax/parse/parser.rs

```diff
@@ -5187,11 +5207,16 @@ impl<'a> Parser<'a> {
             token::CloseDelim(token::Brace) => {}
             _ => {
                 let span = self.span;
-                let token_str = self.this_token_to_string();
-                return Err(self.span_fatal_help(span,
-                                     &format!("expected `,`, or `}}`, found `{}`",
-                                             token_str),
-                                     "struct fields should be separated by commas"))
+                if self.is_doc_comment(span) {
```

Comment:
Looking through the list of tokens again, we actually have `token::DocComment` which may be more applicable than adding a new method here